metatensor-torch/include/metatensor/torch/atomistic/model.hpp

```diff
@@ -288,6 +288,11 @@ class METATENSOR_TORCH_EXPORT ModelMetadataHolder: public torch::CustomClassHold
     void validate() const;
 };
 
+/// Read the metadata of the exported model at the given `path` and return it
+/// as a string. This is useful to check the metadata of a model without
+/// loading it.
+METATENSOR_TORCH_EXPORT std::string read_model_metadata(std::string path);
```

Comment:
I would parse the JSON string as well

```suggestion
METATENSOR_TORCH_EXPORT ModelMetadata read_model_metadata(std::string path);
```